distributed/cli/dask_worker.py

```diff
@@ -407,16 +407,16 @@ def del_pid_file():
     loop = IOLoop.current()
 
     worker_class = import_term(worker_class)
+
+    port_kwargs = _apportion_ports(worker_port, nanny_port, n_workers, nanny)
+    assert len(port_kwargs) == n_workers
+
     if nanny:
         kwargs["worker_class"] = worker_class
         kwargs["preload_nanny"] = preload_nanny
-
-    if nanny:
-        kwargs.update({"worker_port": worker_port, "listen_address": listen_address})
+        kwargs["listen_address"] = listen_address
         t = Nanny
     else:
-        if nanny_port:
-            kwargs["service_ports"] = {"nanny": nanny_port}
```

Comment:
This looks like legacy cruft. service_ports is a property and can't be passed to ``Worker.__init__``.